renderer.h

```diff
@@ -1,14 +1,30 @@
 #pragma once
 
 #include "indexbuffer.h"
-#include "shader.h"
 #include "renderutils.h"
+#include "shader.h"
+#include "texture.h"
 #include "vertexarray.h"
 
+#include <memory>
+
 class Renderer
 {
 private:
+    static std::unique_ptr<Renderer>sInstance;
+
+protected:
+    std::unique_ptr<Shader>mShader { nullptr };
```

Comment:
Possibly a code smell that this is protected?

Reply:
Hmm I thought I was having some issues if I didn't have it as protected. Must have fixed them since then. Switched it to private